connect/api/src/main/java/org/apache/kafka/connect/data/SchemaBuilder.java

```diff
@@ -181,7 +181,7 @@ public SchemaBuilder doc(String doc) {
 
     @Override
     public Map<String, String> parameters() {
-        return Collections.unmodifiableMap(parameters);
+        return parameters == null ? null: Collections.unmodifiableMap(parameters);
```

Comment:
@ijuma The interface docs specify `null` if there are no parameters. I suppose we could do either, although ideally we avoid adding noise to derived schemas by filtering the null entirely (e.g. in the generated Avro schema, don't include unnecessary extra fields just to track that there are no extra schema parameters).